salt/modules/napalm_mod.py

```diff
@@ -1015,8 +1015,19 @@ def pyeapi_nxos_api_args(**prev_kwargs):
     kwargs["username"] = napalm_opts["USERNAME"]
     kwargs["password"] = napalm_opts["PASSWORD"]
     kwargs["timeout"] = napalm_opts["TIMEOUT"]
-    kwargs["transport"] = optional_args.get("transport")
-    kwargs["port"] = optional_args.get("port")
+    # kwargs["transport"] = optional_args.get("transport", "https")
+    # kwargs["port"] = optional_args.get("port", 443)
+
+    if "transport" in optional_args and optional_args["transport"]:
+        kwargs["transport"] = optional_args["transport"]
+    else:
+        kwargs["transport"] = "https"
+
+    if "port" in optional_args and optional_args["port"]:
```

Comment:
**suggestion:**
`if optional_args.get("port"):`

Reply:
using get("key") is generally in case the key doesn't exist, but the code already just checked that it is, hence can use the indexed method. Part of the previous problem was the code was finding the 'port' key (which was None) and just returning that, and using get("key") will do the same with giving you None